### PR TITLE
Feat #533 make usage of sdk session service

### LINKED ISSUE #533: SessionService: Make use of the SessionService implemented in the CDP4Web
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of the COMET-WEB
- [x] I have searched [open](https://github.com/RHEAGROUP/COMET-WEB-Community-Edition/issues) and [closed](https://github.com/RHEAGROUP/COMET-WEB-Community-Edition/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description

A sessionService is now implemented in the CDP4Web library. This SessionService should replace the current one. 
While resolving this, make sure to also bring required features to the CDP4Web implementation (like support of multiple Iteration)

*Edit: Make the current SessionService inherits from the one implemented in CDP4Web, refactoring the current one and making sure to keep required features (e.g., file uploads in UpdateThings method)
### Logs

  - [ ] There are no logs regarding the issue
  - [ ] I have attached a log file or included the relevant section in this issue  
  
### System Configuration

- COMET version:
  - [ ] COMET-WEB: 
  - [ ] Other: 
- Web Services:
  - [ ] COMET Web Services version:

### PR DESCRIPTION
**Pull request by joao4all**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/COMET-WEB-Community-Edition/pulls) open
- [x] I have verified that I am following the COMET-WEB [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/COMET-WEB-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description
Closes #533 SessionService: Make use of the SessionService implemented in the CDP4Web

- Some components, services and view models were refactored in order to be compliant with the sdk session service

### CODE REVIEW

File: COMET.Web.Common/Services/SessionManagement/SessionService.cs [hunk 1]
```diff
@@ -609,6 +225,46 @@ public Task<Result> ReadEngineeringModels(IEnumerable<EngineeringModelSetup> eng
             return this.ReadEngineeringModels(engineeringModelSetups.Select(x => x.EngineeringModelIid));
         }
 
+        /// <summary>
+        /// Writes an <see cref="OperationContainer" /> to the <see cref="ISession" />
+        /// </summary>
+        /// <param name="operationContainer">The <see cref="OperationContainer" /> to write</param>
+        /// <param name="files">A <see cref="IReadOnlyCollection{T}"/> of the file paths as <see cref="string"/> to create or update</param>
+        /// <returns>A <see cref="Task{T}" /> with the <see cref="Result" /> of the operation</returns>
+        public async Task<Result> WriteTransaction(OperationContainer operationContainer, IReadOnlyCollection<string> files)
```
Comment (antoineatrhea) [inline COMET.Web.Common/Services/SessionManagement/SessionService.cs:234]: would be nice to move that one in the SDK. Not for now but please open a Ticket on the SDK for it

File: COMETwebapp.Tests/Components/SystemRepresentation/SystemRepresentationBodyTestFixture.cs [hunk 2]
```diff
@@ -82,10 +82,13 @@ public void SetUp()
             this.session = new Mock<ISession>();
             this.messageBus = new CDPMessageBus();
 
-            this.sessionService = new SessionService(logger.Object, this.messageBus)
+            this.sessionService = new Mock<ISessionService>();
+            this.sessionService.Setup(x => x.Session).Returns(this.session.Object);
+
+          /*  this.sessionService = new SessionService(logger.Object, this.messageBus)
```
Comment (antoineatrhea) [inline COMETwebapp.Tests/Components/SystemRepresentation/SystemRepresentationBodyTestFixture.cs:88]: commented code

File: COMET.Web.Common/Services/SessionManagement/SessionService.cs [hunk 3]
```diff
@@ -148,7 +116,7 @@ public async Task<Result> ReadIteration(IterationSetup iterationSetup, DomainOfE
                 var openedIteration = this.Session.OpenIterations.FirstOrDefault(x => x.Key.Iid == iterationSetup.IterationIid).Key;
                 this.OpenIterations.Add(openedIteration);
 
-                this.messageBus.SendMessage(SessionStateKind.IterationOpened);
```
Comment (antoineatrhea) [inline COMET.Web.Common/Services/SessionManagement/SessionService.cs:151]: the ReadIteration could make use of the OpenIteration of the SDK

**Review by antoineatrhea — CHANGES_REQUESTED** [review]

File: COMET.Web.Common/Services/SessionManagement/SessionService.cs (hunk 1 above)
Comment (joao4all) [reply COMET.Web.Common/Services/SessionManagement/SessionService.cs:234]: Done

https://github.com/RHEAGROUP/COMET-SDK-Community-Edition/issues/332

**Review by antoineatrhea — APPROVED** [review]

**Pull request merged**